### PR TITLE
Refactoring confirmation, removing duplicate links

### PR DESCRIPTION
**Pull request by dparker1005**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Fixes issue where confirmation page links can sometimes show twice. Also removes duplicate code in the template.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: pages/confirmation.php [hunk 1]
```diff
@@ -10,84 +10,81 @@
...
-		<?php if( !empty( $current_user->membership_level->expiration_period ) && $current_user->membership_level->expiration_period == 'Hour' && apply_filters( 'pmpro_confirmation_display_hour_expiration', true, $current_user ) ){ ?>
-		<li><strong><?php esc_html_e('Expires In', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->membership_level->expiration_number . ' ' . pmpro_translate_billing_period( $current_user->membership_level->expiration_period, $current_user->membership_level->expiration_number ) ); ?></li>
-		<?php }
+	<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav' ) ); ?>">
+		<?php
+		if ( ! empty( $current_user->membership_level ) ) {
+			if ( empty( $confirmation_invoice ) ) { // The invoice already shows a link to the account page.
+				?>
+				<a href="<?php echo esc_url( pmpro_url( 'account' ) ); ?>"><?php esc_html_e( 'View Your Membership Account &rarr;', 'paid-memberships-pro' ); ?></a>
```
Comment (kimcoleman) [inline pages/confirmation.php:82]:
```suggestion
				<span class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav-right' ) ); ?>"><a href="<?php echo esc_url( pmpro_url( 'account' ) ); ?>"><?php esc_html_e( 'View Your Membership Account &rarr;', 'paid-memberships-pro' ); ?></a></span>
```

File: pages/confirmation.php [hunk 2]
```diff
@@ -10,84 +10,81 @@
...
-
-		echo wp_kses_post( $confirmation_message );
 	?>
-	<ul>
-		<li><strong><?php esc_html_e('Account', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->display_name );?> (<?php echo esc_html( $current_user->user_email );?>)</li>
-		<li><strong><?php esc_html_e('Membership Level', 'paid-memberships-pro' );?>:</strong> <?php if(!empty($current_user->membership_level)) echo esc_html( $current_user->membership_level->name ); else esc_html_e("Pending", 'paid-memberships-pro' );?></li>
-		<?php if( !empty( $current_user->membership_level->expiration_period ) && $current_user->membership_level->expiration_period == 'Hour' && apply_filters( 'pmpro_confirmation_display_hour_expiration', true, $current_user ) ){ ?>
-		<li><strong><?php esc_html_e('Expires In', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->membership_level->expiration_number . ' ' . pmpro_translate_billing_period( $current_user->membership_level->expiration_period, $current_user->membership_level->expiration_number ) ); ?></li>
-		<?php }
+	<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav' ) ); ?>">
```
Comment (kimcoleman) [inline pages/confirmation.php:77]: This empty <p> is still showing when there is no content to show (I have a membership and an invoice). So I think this need a bit more architecting. Perhaps we could move the message regarding inactive membership level to its own paragraph. We don't need to use the actions nav p for this.

<?php if ( ! empty( $current_user->membership_level ) && empty( $confirmation_invoice ) ) { ?>
<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav' ) ); ?>">
<span class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav-right' ) ); ?>"><a href="<?php echo esc_url( pmpro_url( 'account' ) ); ?>"><?php esc_html_e( 'View Your Membership Account &rarr;', 'paid-memberships-pro' ); ?></a></span>
</p>
<?php } ?>

<?php if ( empty( $current_user->membership_level ) ) { ?>
<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_membership_pending' ) ); ?>"><?php esc_html_e( 'If your account is not activated within a few minutes, please contact the site owner.', 'paid-memberships-pro' ); ?></p>
<?php } ?>

Something like this... ?

File: pages/confirmation.php [hunk 3]
```diff
@@ -10,84 +10,81 @@
...
+		$confirmation_invoice->getUser();
+		$confirmation_invoice->getMembershipLevel();
+		$confirmation_message .= '<p>' . sprintf( __( 'Below are details about your membership account and a receipt for your initial membership invoice. A welcome email with a copy of your initial membership invoice has been sent to %s.', 'paid-memberships-pro' ), $current_user->user_email ) . '</p>';
+	} else {
+		$confirmation_message .= '<p>' . sprintf( __( 'Below are details about your membership account. A welcome email has been sent to %s.', 'paid-memberships-pro' ), $current_user->user_email ) . '</p>';
+	}
 
-		echo wp_kses_post( $confirmation_message );
+	/**
+	 * All devs to filter the confirmation message.
```
Comment (kimcoleman) [inline pages/confirmation.php:50]:
```suggestion
	 * Allow devs to filter the confirmation message.
```

File: pages/confirmation.php [hunk 4]
```diff
@@ -10,84 +10,81 @@
...
+	$confirmation_invoice = ( ! empty( $pmpro_invoice ) && ! empty( $pmpro_invoice->id ) ) ? $pmpro_invoice : false;
 
-		/**
-		 * All devs to filter the confirmation message.
-		 * We also have a function in includes/filters.php that applies the the_content filters to this message.
-		 * @param string $confirmation_message The confirmation message.
-		 * @param object $pmpro_invoice The PMPro Invoice/Order object.
-		 */
-		$confirmation_message = apply_filters("pmpro_confirmation_message", $confirmation_message, $pmpro_invoice);
+	// Add some details to the confirmation messge about the invoice.
```
Comment (kimcoleman) [inline pages/confirmation.php:40]:
```suggestion
	// Add some details to the confirmation message about the invoice.
```

**Review by kimcoleman — CHANGES_REQUESTED** [review]

File: pages/confirmation.php (hunk 2 above)
Comment (dparker1005) [reply pages/confirmation.php:77]: Agreed, I was just trying to keep the code functionality as close to what was already here. So should we get rid fo that `<p>` tag around the `if` statement and instead wrap the content within the `if`s in paragraph tags?

**Pull request merged**